src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,111 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double RESIZE_COEFFICIENT = 1.5;
+    private Object[] array;
+    private int size;
+
+    public ArrayList(int capacity) {
+        if (capacity > 0) {
+            array = new Object[capacity];
+        } else {
+            array = new Object[DEFAULT_CAPACITY];
+        }
+    }
+
+    public ArrayList() {
+        array = new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size + 1 > array.length) {
+            resizeArray(array);
+        }
+        array[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index == size) {
+            add(value);
+            return;
+        }
+        checkIndex(index);
+        // case when added value into the middle of array:
+        Object lastElement = array[size - 1];
+        System.arraycopy(array, index, array, index + 1, size - index - 1);
+        set(value, index);
+        add((T) lastElement);
```

Comment:
Better add  resizeArray check and do it without **lastElement** 
```suggestion
        System.arraycopy(array, index, array, index + 1, size - index );
        array[index] = value
        size++;
```